pkg/scraper/client/resource/client.go

```diff
@@ -107,17 +107,20 @@ func (kc *kubeletClient) getMetrics(ctx context.Context, url, nodeName string) (
 	if response.StatusCode != http.StatusOK {
 		return nil, fmt.Errorf("request failed, status: %q", response.Status)
 	}
-	b := kc.buffers.Get().([]byte)
+	bp := kc.buffers.Get().(*[]byte)
+	b := *bp
+	defer func() {
+		*bp = b
+		kc.buffers.Put(bp)
+	}()
```

Comment:
deferring something that is not supposed to be deferred can lead to unexpected behaviors, so I'd prefer sticking to the original way

Reply:
Why isn't it supposed to be deferred? Whether the code fails or not, we want to reuse the buffer: those slices are valid for the next request.